paf-mvp-frontend/src/containers/notification/Notification.tsx

```diff
@@ -0,0 +1,65 @@
+import { h } from 'preact';
+import { Done } from '../../components/svg/done/Done';
+import { Attention } from '../../components/svg/attention/Attention';
+import { ISnackBarProps, SnackBar } from '../../components/snack-bar/SnackBar';
+import { useEffect } from 'react';
+import { NotificationEnum } from '../../enums/notification.enum';
+
+export interface INotificationProps {
+  type: NotificationEnum;
+  destroy?: () => void
+}
+
+const TIME_TO_DISPLAY = 10_000;
+
+export const Notification = ({ type, destroy }: INotificationProps) => {
+  const brandName = window.location.hostname;
+  useEffect(() => {
+    setTimeout(() => destroy(), TIME_TO_DISPLAY);
+  }, []);
+
+  const launchPrompt = (event: MouseEvent) => {
+    event.preventDefault();
+    // TODO: launch paf-lib function with callback
+    window.__promptConsent();
+    destroy();
+  };
+
+  let notificationData: Pick<ISnackBarProps, 'icon' | 'title' | 'message'> = {
```

Comment:
OPT: to me, it would be clearer if this block would be in the `default:` case in the `switch`. Here you are initializing a value that will be replaced if `type` is `NotificationEnum.personalized` or `NotificationEnum.default`.

What's even more confusing (and might be explained with a comment) is how we end up in this case: since there is a `NotificationEnum.default`, I expect type to never be different.

In other words, if my understanding is correct, maybe this code would work, and be more comprehensive for me:

```javascript
  let notificationData: Pick<ISnackBarProps, 'icon' | 'title' | 'message'>;

  switch (type) {
    case NotificationEnum.personalized:
      notificationData = {
        icon: <Done />,
        title: `You chose to see personalized content and relevant ads on ${brandName}`,
        message: <div>
          Turn on <a href="#" onClick={launchPrompt}>personalized marketing</a> at any time to make your content and
          ads more relevant on this website.
        </div>
      }
      break;
    case NotificationEnum.default:
    default:
      notificationData = {
        icon: <Attention/>,
        title: `You chose to see standard content and ads on ${brandName}`,
        message: <div>
          Turn on <a href="#" onClick={launchPrompt}>personalized marketing</a> at any time to make your content and
          ads more relevant on this website.
        </div>
      }
      break;
  }
  ```

It would also reduce duplication of the default title.

Does that make sense?

Reply:
Sure. Good point of view